src/jit/morph.cpp

```diff
@@ -12453,22 +12445,12 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
                     }
 
                     /* Special case: "x |= -1" and "x &= 0" */
-                    if  (cmop == GT_AND || cmop == GT_OR)
+                    if (((cmop == GT_AND) && op2->gtOp.gtOp2->IsZero()) ||
+                        ((cmop == GT_OR)  && op2->gtOp.gtOp2->IsIMinusOne()))
```

Comment:
Another instance of liberal condition and making the assumption that we never generate certain IR nodes?